.github/workflows/test-template.yml

```diff
@@ -0,0 +1,1368 @@
+name: CI Template job
+
+on:
+  workflow_call:
+    inputs:
+      reason:
+        description: |
+          Reason why this workflow was called. Could be "buildpulse" for
+          flaky tests detection or "daily-test" for scheduled full builds,
+          "CI" for normal pushes/PRs
+        type: string
+        required: false
+        default: CI
+      ubuntuRunner:
+        description: Override default runner used for linux tests. Can be used to opt-out of buildjet
+        type: string
+        default: buildjet-4vcpu-ubuntu-2004
+      jobTimeout:
+        description: Timeout for the jobs. Default is 35 minutes
+        type: number
+        default: 35
+      jobs:
```

Comment:
Nit: potentially confusing for people later that this has the same name has the regular GitHub "jobs" definition.
Maybe renaming to `jobsToRun` helps.